### PR TITLE
update MSBuild settings to make samples runnable

### PR DESCRIPTION
**Pull request by anurse**
/cc @davidfowl @natemcmaster

### CODE REVIEW

File: samples/SocialWeather/SocialWeather.csproj [hunk 1]
```diff
@@ -1,12 +1,20 @@
...
   <PropertyGroup>
     <TargetFramework>netcoreapp1.1</TargetFramework>
     <PreserveCompilationContext>true</PreserveCompilationContext>
     <OutputType>Exe</OutputType>
+    <RuntimeFrameworkVersion>1.2.0-*</RuntimeFrameworkVersion>
+    <IsPackable>false</IsPackable>
   </PropertyGroup>
 
   <ItemGroup>
+    <Content Include="wwwroot\index.html" />
```
Comment (anurse) [inline samples/SocialWeather/SocialWeather.csproj]: Uhh... weird. Cleaning this up. I renamed from VS, so there might be a bug here.

File: samples/ChatSample/ChatSample.csproj [hunk 2]
```diff
@@ -1,11 +1,15 @@
...
+
+  <Import Project="..\..\build\common.props" />
 
   <PropertyGroup>
     <TargetFramework>netcoreapp1.1</TargetFramework>
     <PreserveCompilationContext>true</PreserveCompilationContext>
     <OutputType>Exe</OutputType>
     <UserSecretsId>aspnet-ChatSample-f11cf018-e0a8-49fa-b749-4c0eb5c9150b</UserSecretsId>
     <PackageTargetFallback>$(PackageTargetFallback);portable-net45+win8</PackageTargetFallback>
+    <RuntimeFrameworkVersion>1.2.0-*</RuntimeFrameworkVersion>
```
Comment (davidfowl) [inline samples/ChatSample/ChatSample.csproj]: This is so that things work once the feature is fixed?

File: samples/SocialWeather/Startup.cs [hunk 3]
```diff
@@ -35,7 +35,7 @@ public void Configure(IApplicationBuilder app, IHostingEnvironment env, ILoggerF
             }
 
             app.UseSockets(o => { o.MapEndpoint<SocialWeatherEndPoint>("/weather"); });
-            app.UseStaticFiles();
+            app.UseFileServer();
```
Comment (anurse) [inline samples/SocialWeather/Startup.cs]: This just makes the sample a little more runnable. Right now it 404s when you Ctrl-F5 it. I could have changed the launch URL but this seemed smoother.

File: samples/ChatSample/ChatSample.csproj [hunk 4]
```diff
@@ -1,11 +1,15 @@
...
+    <IsPackable>false</IsPackable>
```
Comment (davidfowl) [inline samples/ChatSample/ChatSample.csproj]: @natemcmaster why wouldn't they just infer this for `<OutputType>Exe</OutputType>`?

File: samples/SocialWeather/Startup.cs [hunk 5]
```diff
@@ -35,7 +35,7 @@ public void Configure(IApplicationBuilder app, IHostingEnvironment env, ILoggerF
             }
 
             app.UseSockets(o => { o.MapEndpoint<SocialWeatherEndPoint>("/weather"); });
-            app.UseStaticFiles();
```
Comment (moozzyk) [inline samples/SocialWeather/Startup.cs]: why is this needed?

File: samples/SocialWeather/SocialWeather.csproj (hunk 1 above)
Comment (anurse) [inline samples/SocialWeather/SocialWeather.csproj]: Indeed. And @natemcmaster we still need to set it, or something. Either way, it seems appropriate to have it.

**Comment by moozzyk** [comment]
Microsoft.AspNetCore.SignalR.Test.Server is broken too

File: samples/SocialWeather/Startup.cs (hunk 5 above)
Comment (anurse) [inline samples/SocialWeather/Startup.cs]: See below comment. It just makes Ctrl-F5 smoother. `UseFileServer` activates default document, so `index.html` works.

File: samples/ChatSample/ChatSample.csproj (hunk 4 above)
Comment (moozzyk) [inline samples/ChatSample/ChatSample.csproj]: This is even more confusing - if you target both netcoreapp1.1 and net46 what does actually Output type means?

**Comment by anurse** [comment]
Shouldn't the build have caught that? We use the server in our tests, right?

File: samples/SocialWeather/Startup.cs (hunk 3 above)
Comment (moozzyk) [inline samples/SocialWeather/Startup.cs]: I thought 404 was because there was no index.html. I saw you changed that but I don't know why it requires FileServer...

File: samples/SocialWeather/Startup.cs (hunk 5 above)
Comment (moozzyk) [inline samples/SocialWeather/Startup.cs]: I did not know that so 👍

**Comment by moozzyk** [comment]
Test server currently is used only for end-to-end ts client tests which are not run by default since they need to be run from browser.

File: samples/ChatSample/ChatSample.csproj (hunk 4 above)
Comment (natemcmaster) [inline samples/ChatSample/ChatSample.csproj]: @davidfowl IsPackable is true by default on all project types. Not sure why NuGet decided to make it that way. 

Setting this to false means calling "dotnet pack" on the .sln file won't try to pack this project.

Comment (natemcmaster) [inline samples/ChatSample/ChatSample.csproj]: Also, there are valid scenarios for packing OutputType=Exe. e.g. dotnet-cli tools packages.

File: samples/ChatSample/ChatSample.csproj (hunk 2 above)
Comment (natemcmaster) [inline samples/ChatSample/ChatSample.csproj]: To be clear until https://github.com/dotnet/sdk/pull/820 is in, this will break apps if you didn't also reference Internal.AspNetCore.Sdk.

File: samples/ChatSample/ChatSample.csproj (hunk 4 above)
Comment (moozzyk) [inline samples/ChatSample/ChatSample.csproj]: Sure but in this case (which I think is more common) it probably means nothing but introduces confusion.

Comment (natemcmaster) [inline samples/ChatSample/ChatSample.csproj]: > introduces confusion

Ok. Add a comment.

File: test/Microsoft.Extensions.WebSockets.Internal.Tests/WebSocketConnectionTests.PingPong.cs [hunk 6]
```diff
@@ -29,7 +29,7 @@ public class PingPongBehavior
 
                     // Act
                     // Wait for pings to be sent
-                    await Task.Delay(200);
+                    await Task.Delay(500);
```
Comment (mikaelm12) [inline test/Microsoft.Extensions.WebSockets.Internal.Tests/WebSocketConnectionTests.PingPong.cs]: 🙁
